Automattic/jetpack
# Protect card: Add the Auto-firewall status column (with tooltip)

This PR populates the "Auto-firewall" section on the My Jetpack Protect card with the current auto-firewall status from Protect.

See Design P2: p1HpG7-rFA-p2 (Figma link is within the post)

Screenshots are shown below in the testing instructions.






Note: The "Logins Blocked" column is not functional yet.  It's currently displaying 0 only as a placeholder. The functionality will be added in a followup PR.

## Proposed changes:

- Load the `Waf_Runner` package into the My Jetpack initializer and output's the Waf config data into window.MyJetpackInitialState.
- Display the current "Auto-firewall" status within the Protect product card in My Jetpack (based on current product state).
- Display "0" as a placeholder in the "Logins Blocked" column (just for now, until follow-up PR).

### Other information:


## Jetpack product discussion



Project Thread: pbNhbs-aP6-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






*** You can reference the link to the Figma design in the design P2 post: p1HpG7-rFA-p2

- Create a Jurassic Ninja site, with the Beta Tester plugin, running this branch on the Jetpack plugin (only the Jetpack plugin for now. We'll run the branch on Protect in later steps).
- Go to My Jetpack and view the Protect product card. Verify that the "Auto-firewall" column says "Off" with the appropriate icon.  (See screenshot below:)
![Screen Shot on 2024-07-16 at 23-19-38 (1)](https://github.com/user-attachments/assets/a1745fb1-a975-46e7-b278-360d4a922da4)
- In the Beta Tester plugin, add this branch to the Protect plugin.
- Open the Protect page. Select the Free plan, if prompted.
- Go back to My Jetpack. Verify The Auto-firewall status is now "inactive" with the appropriate icon, and has an info tooltip. (See screenshot(s) below:)
![Screen Shot on 2024-07-16 at 23-28-55](https://github.com/user-attachments/assets/ba141472-8a3f-4891-b90c-8adcd342bf11)

![Screen Shot on 2024-07-16 at 23-30-02](https://github.com/user-attachments/assets/2e55f032-3cb5-4225-8787-1d92c4d33d2d)

- Go back to the Protect page, and click the "Firewall" tab.
- Click the "Upgrade to enable automatic firewall protection" button. Complete checkout (purchase with credits).
- Once purchase is completed, go to My Jetpack. Verify The Auto-firewall status is "inactive", but the tooltip is Not showing. See Screenshot:
![Screen Shot on 2024-07-16 at 23-32-23](https://github.com/user-attachments/assets/82212911-681c-46f4-b260-e2f044d989e8)

- Go to the Protect Firewall page, click the toggle to "Enable automatic firewall protection".
- Go back to My Jetpack and verify the "Auto-firewall" status is "On" with the appropriate icon. (See screenshot below:)
![Screen Shot on 2024-07-16 at 23-32-54](https://github.com/user-attachments/assets/6b883053-f99a-46f7-87ca-2cfba54eab3a)
- Now view the My Jetpack product cards while contracting & expanding the viewport width. Verify the product cards do not get smaller width than 300px and that all cards wrap correctly (number of columns are adjusted smoothly and appropriately).
